packages/manager/apps/container/src/payment-modal/PaymentModal.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState, useEffect } from 'react';
+import { Modal } from 'react-bootstrap';
+import { useTranslation } from 'react-i18next';
+
+import { useShell } from '@/context';
+
+import { fetchIcebergV6 } from '@ovh-ux/manager-core-api';
+
+import {
+  NO_DEFAULT_CARD,
+  EXPIRED_CARD,
+  SOON_EXPIRED_CARD
+} from './constants';
+
+import './styles.scss';
+
+interface IPaymentMethod {
+  icon?: any;
+  label: string;
+  status: string;
+  default: boolean;
+  oneclick: boolean;
+  lastUpdate: string;
+  description: string;
+  integration: string;
+  paymentType: string;
+  creationDate: string;
+  paymentMeanId: number;
+  expirationDate?: string;
+  paymentSubType: string;
+  paymentMethodId: number;
+}
+
+interface IPaymentMethodResponse {
+  data: IPaymentMethod[];
+}
+
+const PaymentModal = (): JSX.Element => {
+  const [mode, setMode] = useState('');
+  const { t } = useTranslation('payment-modal');
+  const [showPaymentModal, setShowPaymentModal] = useState(false);
+  const shell = useShell();
+
+  const paymentMethodURL = shell
+    .getPlugin('navigation')
+    .getURL('dedicated', '#/billing/payment/method');
+
+  useEffect(() => {
+    const getPaymentMethods = async () => {
+      const paymentResponse: IPaymentMethodResponse = await fetchIcebergV6({ route: '/me/payment/method' });
+      const hasDefaultPaymentMethod = paymentResponse.data && paymentResponse.data.length > 0 && paymentResponse.data.find(currentPaymentMethod => currentPaymentMethod.default);
```

Comment:
**nitpick:** We can directly use filtering available on API `/me/payment/method` for `default` and ` paymentType`

Reply:
@anooparveti I don't think it's adapted here as we need to check either "there is no payment method" (need to check for all types) or "a credit card has expired or is about to" (but not necessary the default one)